great_expectations_cloud/agent/agent.py

```diff
@@ -180,7 +181,8 @@ def _handle_event_as_thread_enter(self, event_context: EventContext) -> None:
             # request that this message is redelivered later. If the event is UnknownEvent
             # we don't understand it, so requeue it in the hope that someone else does.
             loop = asyncio.get_event_loop()
-            loop.create_task(event_context.redeliver_message())
+            # store a reference the task to ensure it isn't garbage collected
+            self._redeliver_msg_task = loop.create_task(event_context.redeliver_message())
```

Comment:
https://docs.astral.sh/ruff/rules/asyncio-dangling-task/
>Per the asyncio documentation, the event loop only retains a weak reference to tasks. If the task returned by `asyncio.create_task` and `asyncio.ensure_future` is not stored in a variable, or a collection, or otherwise referenced, it may be garbage collected at any time. This can lead to unexpected and inconsistent behavior, as your tasks may or may not run to completion.